docs/api.rst

```diff
@@ -0,0 +1,17 @@
+Reactive API Documentation
+--------------------------
+
+
+.. toctree::
+
+    charms.reactive.decorators
+    charms.reactive.helpers
+    charms.reactive.relations
+    charms.reactive.bus
+    charms.reactive.cli
```

Comment:
Consider only having the actual API in the documentation. Does charms.reactive.bus need to be here encouraging people to use internal only functions? Are these internal only functions? Do we need to separate internals from API better?